### ISSUE TITLE
assets:install Warning: Illegal offset type in isset or empty

### ISSUE THREAD

**Issue by rayrigam** [sensiolabs/SensioDistributionBundle#268]
After a `composer update` when running the `assets:install` command, I get this error:

> [RuntimeException]
>   An error occurred when executing the "'assets:install '\''web'\'''" command
>     [Symfony\Component\Debug\Exception\ContextErrorException]
>     Warning: Illegal offset type in isset or empty

I'm still using Symfony 2.7 with `"sensio/distribution-bundle": "~4.0"`

Any ideas what the problem may be and how to fix it?

The problem occurs in line 175 of Sensio\Bundle\DistributionBundle\Composer\ScriptHandler

`static::executeCommand($event, $consoleDir, 'assets:install '.$symlink.escapeshellarg($webDir), $options['process-timeout']);`

**Comment by xabbuh** [comment]
Can you show the contents of your `composer.json` file as well as the output of `composer show`?

**Comment by rayrigam** [comment]
The require part of composer.json looks like this:
```
"require": {
        "php": ">=5.3.9",
        "ext-gd": "*",
        "ext-intl": "*",
        "ext-memcached": "*",
        "ext-mbstring": "*",
        "symfony/symfony": "2.7.*",
        "twig/extensions": "~1.0",
        "symfony/assetic-bundle": "~2.3",
        "symfony/swiftmailer-bundle": "~2.3",
        "symfony/monolog-bundle": "~2.4",
        "sensio/distribution-bundle": "~4.0",
        "sensio/framework-extra-bundle": "~3.0,>=3.0.2",
        "sensio/generator-bundle": "~2.3",
        "incenteev/composer-parameter-handler": "~2.0",

        "propel/propel-bundle": "1.4.*@dev",
        "willdurand/propel-typehintable-behavior": "1.0.*@dev",
        "friendsofsymfony/user-bundle": "2.0.*@dev",
        "friendsofsymfony/elastica-bundle": "3.1.*@dev",
        "aws/aws-sdk-php": "3.0.*@dev",
        "imagine/imagine": "0.7.*@dev"
    },
```

The output of composer is:

> Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::installAssets
>   [Symfony\Component\Debug\Exception\ContextErrorException]
>   Warning: Illegal offset type in isset or empty
> Script Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::installAssets handling the post-update-cmd event terminated with an exception
> [RuntimeException]
> An error occurred when executing the "'assets:install '\''web'\'''" command
> [Symfony\Component\Debug\Exception\ContextErrorException]
> Warning: Illegal offset type in isset or empty

When I run this command `$ php app/console assets:install --verbose` , I get:

> [Symfony\Component\Debug\Exception\ContextErrorException]
>   Warning: Illegal offset type in isset or empty
> Exception trace:
>  () at /vagrant/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/ReplaceAliasByActualDefinitionPass.php:133
>  Symfony\Component\Debug\ErrorHandler->handleError() at /vagrant/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/ReplaceAliasByActualDefinitionPass.php:133
>  Symfony\Component\DependencyInjection\Compiler\ReplaceAliasByActualDefinitionPass->updateFactoryReferenceId() at /vagrant/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/ReplaceAliasByActualDefinitionPass.php:80
>  Symfony\Component\DependencyInjection\Compiler\ReplaceAliasByActualDefinitionPass->process() at /vagrant/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/Compiler.php:104
>  Symfony\Component\DependencyInjection\Compiler\Compiler->compile() at /vagrant/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:578
>  Symfony\Component\DependencyInjection\ContainerBuilder->compile() at /vagrant/app/bootstrap.php.cache:2651
>  Symfony\Component\HttpKernel\Kernel->initializeContainer() at /vagrant/app/bootstrap.php.cache:2429
>  Symfony\Component\HttpKernel\Kernel->boot() at /vagrant/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:70
>  Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /vagrant/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:120
>  Symfony\Component\Console\Application->run() at /vagrant/app/console:27

**Comment by rayrigam** [comment]
The problem is with [friendsofsymfony/elastica-bundle version "3.1.*@dev"](https://github.com/FriendsOfSymfony/FOSElasticaBundle/tree/3.1.x). In the method `updateFactoryReferenceId` of the class `Symfony\Component\DependencyInjection\Compiler\ReplaceAliasByActualDefinitionPass.php`, the second parameter passed `$referenceId` should be a string. Instead, the ElasticaBundle passes an object.
In the first call it passes `fos_elastica.client` and in a second call, when it throws the above mentioned exception (as it cannot be used as an array key in line 133 `isset($replacements[$referenceId])`), it passes this object as `$referenceId`:
```
Symfony\Component\DependencyInjection\Reference Object
(
    [id:Symfony\Component\DependencyInjection\Reference:private] => fos_elastica.client.default
    [invalidBehavior:Symfony\Component\DependencyInjection\Reference:private] => 1
    [strict:Symfony\Component\DependencyInjection\Reference:private] => 1
)
```
To fix this error, I upgraded the friendsofsymfony/elastica-bundle version to "3.2.*@dev". I'm closing this issue as it had to do with elastica-bundle, and was resolved with the new version.
